src/sentry/rules/conditions/event_frequency.py

```diff
@@ -265,19 +294,70 @@ class EventFrequencyCondition(BaseEventFrequencyCondition):
     def query_hook(
         self, event: GroupEvent, start: datetime, end: datetime, environment_id: str
     ) -> int:
-        sums: Mapping[int, int] = self.tsdb.get_sums(
-            model=get_issue_tsdb_group_model(event.group.issue_category),
+        sums: Mapping[int, int] = self.get_sums(
             keys=[event.group_id],
+            group=event.group,
+            model=get_issue_tsdb_group_model(event.group.issue_category),
             start=start,
             end=end,
             environment_id=environment_id,
-            use_cache=True,
-            jitter_value=event.group_id,
-            tenant_ids={"organization_id": event.group.project.organization_id},
             referrer_suffix="alert_event_frequency",
         )
         return sums[event.group_id]
 
+    def get_chunked_sums(
+        self,
+        groups: list[Group],
+        start: datetime,
+        end: datetime,
+        environment_id: str,
+        referrer_suffix: str,
+    ) -> dict[int, int]:
+        batch_sums: dict[int, int] = defaultdict(int)
+        for group_chunk in chunked(groups, SNUBA_LIMIT):
+            group = groups[0]
+            sums = self.get_sums(
+                keys=[group.id for group in groups],
+                group=group,
+                model=get_issue_tsdb_group_model(group.issue_category),
+                start=start,
+                end=end,
+                environment_id=environment_id,
+                referrer_suffix="alert_event_frequency",
+            )
+            batch_sums.update(sums)
+        return batch_sums
+
+    def batch_query_hook(
+        self, group_ids: Sequence[int], start: datetime, end: datetime, environment_id: str
+    ) -> dict[int, int]:
+        batch_sums: dict[int, int] = defaultdict(int)
+        groups = Group.objects.filter(id__in=group_ids)
+        error_issues = [group for group in groups if group.issue_category == GroupCategory.ERROR]
+        generic_issues = [group for group in groups if group.issue_category != GroupCategory.ERROR]
+
+        if error_issues:
+            error_sums = self.get_chunked_sums(
```

Comment:
In javascript i'd generally define `get_chunked_sums` inside of the `batch_query_hook` function and then reference the `start`, `end`, and `environment_id` from the outer scope. Is that a faux pas in python? (if not, it might help DRY this code up a little more  and reduce it by 9 more lines 😎)

It'd be something like:
```python
def get_chunked_sums(groups: list[Group], referrer_suffix: str):
    # same code you have for get_chunked_sums

if error_issues:
    error_sums = get_chunked_sums(
        groups=error_issues,

        # looks like referrer_suffix could be DRY'd up and hardcoded in get_chunked_sums as well
        referrer_suffix="alert_event_frequency"
    )
```